### PR TITLE
Add some docs.

### LINKED ISSUE #11: Add documentation and README.

### PR DESCRIPTION
**Pull request by singulared**
closes #11

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,138 @@
+//! A proxy actor and infrastructure for asynchronous and clear cache interaction
```
Comment (AndreyErmilov) [inline src/lib.rs]: Maybe "A proxy actor and infrastructure for asynchronous and clear cache interaction for Actix -web and actor frameworks"?

File: src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,138 @@
...
+//!     let number = cache
+//!         .send(query.into_cache(&fib))
+//!         .await
+//!         .unwrap()
+//!         .unwrap();
+//!     HttpResponse::Ok().body(format!("Generate Fibonacci number {}", number))
+//! }
+//! ```
+//!
+//! This full example and other examples you can see on [github.com](https://github.com/rambler-digital-solutions/actix-cache/blob/master/examples/actix_web.rs)
```
Comment (AndreyErmilov) [inline src/lib.rs]: I think we should move line 110 with `This full example and other examples ..`  to line 70. This code a little bit unclear without full example.

File: src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,138 @@
...
+//! #     type Result = <GetNumber as Message>::Result;
+//! #
+//! #     fn handle(&mut self, msg: GetNumber, _ctx: &mut Self::Context) -> Self::Result {
+//! #         42
+//! #     }
+//! # }
+//! #
+//! # type Cache = CacheActor<RedisBackend>;
+//! async fn index(
+//!     fib: web::Data<Addr<FibonacciActor>>,
```
Comment (AndreyErmilov) [inline src/lib.rs:98]: What do you think about this example?
```rust
async fn index(
    db: web::Data<Addr<DatabaseActor>>,
    cache: web::Data<Addr<Cache>>
) -> impl Responder {
    let query = GetPosts { limit: 10 };
    let posts = cache
        .send(query.into_cache(&db))
        .await
        .unwrap()
        .unwrap();
    HttpResponse::Ok().body(format!("There are {} posts", posts.len()))
}
```

**Review by AndreyErmilov — COMMENTED** [review]
Looks good 😎

File: src/lib.rs (hunk 1 above)
Comment (singulared) [reply src/lib.rs]: Only for Actix actor framework)

File: src/lib.rs (hunk 3 above)
Comment (singulared) [reply src/lib.rs:98]: Nice. But this example will not be built without a declaration of all necessary types. And can't be a doctest.

File: src/lib.rs (hunk 1 above)
Comment (singulared) [reply src/lib.rs]: More specifically, actix-cache has nothing to do with actix-web. It's just can be used with actix-web.

File: src/lib.rs (hunk 3 above)
Comment (singulared) [reply src/lib.rs:98]: I think we should add a separate example for actix-web + diesel. But it must be a different repository. I don't want to add diesel into dev-dependencies.

File: src/lib.rs (hunk 1 above)
Comment (AndreyErmilov) [reply src/lib.rs]: Ok. Then so "A proxy actor and infrastructure for asynchronous and clear cache interaction for Actix framework."

File: src/lib.rs [hunk 4]
```diff
@@ -1,3 +1,137 @@
...
+//! # A tour of actix-cache
+//!
+//! This crate consist of three main part:
+//! * [Cache] actor.
+//! * [Backend] trait and its implementation ([RedisBackend]).
+//! * [Cacheable] trait.
+//!
+//! ## Features
+//! * Async/Sync cache backend support.
+//! * Dogpile effect prevention.
```
Comment (AndreyErmilov) [inline src/lib.rs]: Maybe we should add a link to an article describing "Dogpile effect"? E.g. http://highscalability.com/blog/2014/7/30/preventing-the-dogpile-effect-problem-and-solution.html

File: src/cache.rs [hunk 5]
```diff
@@ -78,19 +75,53 @@ pub trait Cacheable {
...
+    /// #[rtype(result = "()")]
+    /// struct QueryNothing {
+    ///     id: Option<i32>,
+    /// }
+    ///
+    /// #[actix_rt::main]
+    /// async fn main() {
+    ///     let upstream = Upstream.start();
+    ///     let query = QueryNothing { id: Some(1) };
+    ///     query.into_cache(&upstream);
```
Comment (AndreyErmilov) [inline src/cache.rs]: ```let query: QueryCache<Upstream, QueryNothing > = query.into_cache(&upstream);```

File: src/cache.rs [hunk 6]
```diff
@@ -78,19 +75,53 @@ pub trait Cacheable {
...
-            upstream,
+            upstream: upstream.clone(),
             message: self,
         }
     }
 }
 
+/// Intermediate actix message which handled by Cache actor.
+///
+/// This message a product of upstream message and upstream actor address.
```
Comment (AndreyErmilov) [inline src/cache.rs:124]: It is not clear to me what QueryCache is. So I think we should add ```QueryCache is a struct that includes base message with user data and address of an actor that is a recipient of this message. You can only send QueryCache messages to Cache actor.```

**Review by AndreyErmilov — APPROVED** [review]

File: src/lib.rs (hunk 3 above)
Comment (singulared) [reply src/lib.rs:98]: #14

**Pull request merged**